lib/carbon/cache.py

```diff
@@ -42,6 +45,10 @@ def gen_queue(self):
 
   def store(self, metric, datapoint):
     self.size += 1
+    dbFilePath = getFilesystemPath(metric)
+    if settings.CACHE_CREATE_WHISPER_FILES:
+      if (len(self[metric]) == 0):
+        queueWhisperCreate(metric, dbFilePath)
     self[metric].append(datapoint)
```

Comment:
Maybe it makes sense to add this to the cache *only* if the file already exists, to avoid a race condition later? This suggestion feels iffy for some reason, though.